packages/cashier/src/Components/__tests__/on-ramp-provider-popup.spec.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import OnRampProviderPopup from '../on-ramp-provider-popup';
+
+jest.mock('Stores/connect', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    connect: () => Component => Component,
+}));
+
+jest.mock('@deriv/components', () => ({
+    ...jest.requireActual('@deriv/components'),
+    Loading: () => <div>Loading</div>,
+}));
+
+describe('<OnRampProviderPopup />', () => {
+    const props = {
+        api_error: '',
+        deposit_address: 'tb1qhux20f7h42ya9nqdntl6r9p7p264a2ct8t3n6p',
+        is_deposit_address_loading: false,
+        is_deposit_address_popover_open: false,
+        is_requesting_widget_html: false,
+        selected_provider: {
+            name: 'Changelly',
+            should_show_deposit_address: true,
+            onMountWidgetContainer: jest.fn(),
+        },
+        should_show_dialog: false,
+        should_show_widget: false,
+        widget_error: '',
+        widget_html: 'Widget HTML',
+        onClickCopyDepositAddress: jest.fn(),
+        onClickDisclaimerContinue: jest.fn(),
+        onClickGoToDepositPage: jest.fn(),
+        setIsOnRampModalOpen: jest.fn(),
+    };
+
+    it('should not render <OnRampProviderPopup /> component', () => {
+        const { container } = render(<OnRampProviderPopup {...props} selected_provider={null} />);
+
+        expect(container.querySelector('.on-ramp__popup')).not.toBeInTheDocument();
```

Comment:
Can the element be targeted with `data-testid` rather than class name? Since class names might get modified

Reply:
@likhith-deriv make sense in this case, thanks for the notice

Reply:
fixed